esmvalcore/preprocessor/_regrid.py

```diff
@@ -171,16 +191,20 @@ def parse_cell_spec(spec):
     return dlon, dlat
 
 
-def _generate_cube_from_dimcoords(latdata, londata, circular: bool = False):
+def _generate_cube_from_dimcoords(
+    latdata: ArrayLike,
```

Comment:
On further inspection the use of `ArrayLike` is not right in several places in the module: only `np.ndarray`s (or possibly `da.Array`s) can be used as argument. This doesn't show when you run `mypy` through pre-commit because [there the installed dependencies are very limited](https://github.com/pre-commit/mirrors-mypy):
https://github.com/ESMValGroup/ESMValCore/blob/7a88619623bf76163881dadbbc089cdb51f300eb/.pre-commit-config.yaml#L45-L47
and therefore certain checks cannot be performed. Shall we add `numpy` to that list to at least have numpy related checks run?